libvast/builtins/printers/json.cpp

```diff
@@ -0,0 +1,73 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/arrow_table_slice.hpp>
+#include <vast/concept/printable/vast/json.hpp>
+#include <vast/plugin.hpp>
+
+#include <arrow/record_batch.h>
+
+namespace vast::plugins::json_printer {
+
+class plugin : public virtual printer_plugin {
+public:
+  [[nodiscard]] auto
+  make_printer(const record& options, [[maybe_unused]] type input_schema,
+               const operator_control_plane&) const
+    -> caf::expected<printer_plugin::printer> override {
+    auto input_type = caf::get<record_type>(input_schema);
+    return [input_type, plugin_options = options](
+             generator<table_slice> slices) -> generator<chunk_ptr> {
+      // JSON printer should output NDJSON, see:
+      // https://github.com/ndjson/ndjson-spec
+      auto printer = vast::json_printer{{.oneline = true}};
```

Comment:
Can you add a TODO here that since this printer is per-schema we can write an optimized version of it that gets the schema ahead of time and only expects data corresponding to exactly that schema?